examples/dbscan/ArborX_DBSCAN.hpp

```diff
@@ -120,29 +120,45 @@ void dbscan(ExecutionSpace exec_space, Primitives const &primitives,
 
   clock::time_point start_total;
   clock::time_point start;
-  clock::time_point start_local;
-  std::chrono::duration<double> elapsed_construction;
-  std::chrono::duration<double> elapsed_stat;
-  std::chrono::duration<double> elapsed_neigh;
-  std::chrono::duration<double> elapsed_query;
-  std::chrono::duration<double> elapsed_cluster;
+  std::chrono::duration<double> elapsed_construction = clock::duration::zero();
+  std::chrono::duration<double> elapsed_stat = clock::duration::zero();
+  std::chrono::duration<double> elapsed_neigh = clock::duration::zero();
+  std::chrono::duration<double> elapsed_query = clock::duration::zero();
+  std::chrono::duration<double> elapsed_cluster = clock::duration::zero();
   std::chrono::duration<double> elapsed_total = clock::duration::zero();
-  std::chrono::duration<double> elapsed_verify;
+  std::chrono::duration<double> elapsed_verify = clock::duration::zero();
 
-  start_total = clock::now();
+  auto clock_start = [&exec_space, verbose](clock::time_point &clock_time) {
+    if (verbose)
+    {
+      exec_space.fence();
+      clock_time = clock::now();
+    }
+  };
+  auto clock_stop = [&exec_space,
+                     verbose](clock::time_point const &clock_time,
+                              std::chrono::duration<double> &elapsed) {
```

Comment:
Same comment, I find it a bit convoluted.
Also why do you need to roll your own utility?  Do you know about `Kokkos::Timer`?

Reply:
I'm not introducing any new utility in this patch. Whether `Kokkos::Timer` is better or not, could be discussed separately.

Reply:
Switched to `Kokkos::Timer`.